Zend/zend_hash.c

```diff
@@ -2531,6 +2861,10 @@ ZEND_API void ZEND_FASTCALL zend_hash_sort_ex(HashTable *ht, sort_func_t sort, b
 		return;
 	}
 
+	if (HT_IS_PACKED(ht)) {
+		zend_hash_packed_to_hash(ht); // TODO: ???
```

Comment:
I guess it'd be possible to add an optimization for packed array when `renumber` is true, sorting `sizeof(zval)` and a new function along the lines of `zend_hash_zval_with_extra_renum_swap` 
(and set `Z_EXTRA(ht->arPacked[i]) = i;` to continue to make it a stable sort)

I assume the `TODO: ???` refers to the fact the below HT_IS_PACKED calls are now impossible.
- When renumber is false, would it make sense to declare `const bool was_packed = HT_IS_PACKED(ht);` before calling `zend_packed_to_hash` and use that?

Reply:
This makes sense.